### PR TITLE
[BEAM-4302] add beam dependency checks

### PR DESCRIPTION
**Pull request by yifanzou**
Add a Jenkins job to do beam dependency check and generate a report in html format.
General workflow:
    - :dependencyUpdates (Java check)
    - :beam-sdks-python:dependencyUpdates (Python Check)
    - run python script to extract, analyze and reorganize check results
    - send report to dev@
------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

### CODE REVIEW

**Comment by yifanzou** [comment]
Run Python Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Java Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Java Dependency Check

**Comment by yifanzou** [comment]
Run Python Dependency Check

**Comment by yifanzou** [comment]
Run Java Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

File: .test-infra/jenkins/dependency_check/bigquery_client_utils.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    query_job = self.bigquery_client.query(query)
+    rows = list(query_job)
+    if len(rows) == 0:
+      return None
+    assert len(rows) == 1
+    return rows[0]['release_date']
+
+
+  def add_dep_to_table(self, dep, version, release_date, is_current_using=False):
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/bigquery_client_utils.py]: What is is_current_using ? (also, The name probably has to be slightly changed for clarity. )

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+        if (not curr_minor_ver.isdigit() or not latest_minor_ver.isdigit()) and curr_minor_ver != latest_minor_ver:
+          return True
+        elif int(curr_minor_ver) + _MAX_MINOR_VERSION_DIFF <= int(latest_minor_ver):
+          return True
+  return False
+
+
+def query_dependency_release_dates(bigquery_client, dep_name, curr_ver, latest_ver):
+  """
+  Query release dates of current using version and the latest version from BQ tables.
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: current version and the latest released version

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+    curr_ver: the current using version
+    latest_ver: the later version
+  Return:
+    curr_release_date
+    latest_release_date
+  """
+  try:
+    curr_release_date = bigquery_client.query_dep_release_date(dep_name, curr_ver)
+    latest_release_date = bigquery_client.query_dep_release_date(dep_name,latest_ver)
+    if curr_release_date is None:
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Why would this be None ? First time we are entering data for this dependency ?

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+      bigquery_client.add_dep_to_table(dep_name, curr_ver, get_date(), is_current_using=True)
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Not sure why is_current_using=True here. Can you add a comment ?

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+      bigquery_client.add_dep_to_table(dep_name, latest_ver, get_date(), is_current_using=False)
+      latest_release_date = get_date()
+  except Exception:
+    raise
+  return curr_release_date, latest_release_date
+
+
+def compare_dependency_release_dates(curr_release_date, latest_release_date):
+  """
+  Compare release dates of current using version and the latest version.
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: current version and latest released version

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+    latest_release_date
+  Return:
+    boolean
+  """
+  if curr_release_date is None or latest_release_date is None:
+    return True
+  else:
+    if (latest_release_date - curr_release_date).days >= _MAX_STALE_DAYS:
+      return True
+  return False
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Consider patch version ?

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+
+
+def get_date():
+  """
+  Get the current date in YYYY-MM-DD
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Do we need a function for this ?

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+  return datetime.today().date()
+
+
+def generate_report(file_path, sdk_type, project_id, dataset_id, table_id):
+  """
+  Write SDK dependency check results into a html report.
+  Args:
+    file_path: the path that report will be write into.
+    sdk_type: String [Java, Python, TODO: Go]
+    project_id: the gcloud project ID that used for Bigquery API requests.
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: that is used

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+    subtitle = "<h2>High Priority Dependency Updates Of Beam {} SDK:</h2>\n".format(sdk_type)
+    table_fields = """<tr>
+      <td><b>{0}</b></td>
+      <td><b>{1}</b></td>
+      <td><b>{2}</b></td>
+      <td><b>{3}</b></td>
+      <td><b>{4}</b></td>
+      </tr>""".format("Dependency Name",
+                      "Current Version",
+                      "Later Version",
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Latest version ?

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+                      "Current Version Release Date",
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Release data of currently used version

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 11]
```diff
@@ -0,0 +1,279 @@
...
+                      "Later Version Release Date")
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Data of the latest release

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator_test.py [hunk 12]
```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+#
+#
+#    Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (chamikaramj) [inline .test-infra/jenkins/dependency_check/dependency_check_report_generator_test.py]: Do we actually run these tests ? I couldn't find a reference to this. How about executing these tests as a part of the Jenkins job ?

File: sdks/python/build.gradle [hunk 13]
```diff
@@ -230,3 +230,12 @@ task postCommit() {
   dependsOn "hdfsIntegrationTest"
   dependsOn "postCommitVRTests"
 }
+
+task dependencyUpdates(dependsOn: ':dependencyUpdates') {
+  doLast {
+    exec {
+      executable 'sh'
```
Comment (chamikaramj) [inline sdks/python/build.gradle]: Can you explain why we need this in addition to the script run by the Jenkins job ?

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py (hunk 6 above)
Comment (yifanzou) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Not here. If we decide to consider the patch version, I will add it into "compare_dependency_versions(...)"

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Dependency Check

File: .test-infra/jenkins/job_Dependency_Check.groovy [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+      common_job_properties.setGradleSwitches(delegate)
+      switches('-Drevision=release')
+    }
+
+    shell('cd ' + common_job_properties.checkoutDir +
+            ' && bash .test-infra/jenkins/dependency_check/generate_report.sh')
+  }
+
+  def date = new Date().format('yyyy-MM-dd')
+  publishers {
```
Comment (alanmyrvold) [inline .test-infra/jenkins/job_Dependency_Check.groovy]: do you need to add archiveArtifacts to this publishers clause to get the html archived for viewing later in the build results?

**Comment by yifanzou** [comment]
Run Seed Job

File: .test-infra/jenkins/job_Dependency_Check.groovy (hunk 14 above)
Comment (yifanzou) [reply .test-infra/jenkins/job_Dependency_Check.groovy]: yes, added archiving script to publish the report.

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
@chamikaramj  PTAL
Major changes:
1. Add deletion and cleanup helper methods in the bigquery_client_utils that supports maintaining the dep states in the BQ tables.
2. Add archiveArtifact in the Jenkins job, this will archive the dependency report and allows people to review it via Jenkins UI. 
3. Rename bool variable "is_current_using" --> "is_currently_used", reinitialized bq tables.

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator_test.py (hunk 12 above)
Comment (chamikaramj) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator_test.py]: Why not running these a part of the Jenkins job ? I think if it's not very expensive we should run them (we do the same thing for performance tests). That will make us more confident on the results. If we don't run them somewhere automatically probably no one ever will and tests/code will become stale.

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py [hunk 15]
```diff
@@ -0,0 +1,279 @@
...
+    # compare minor versions
+    else:
+      curr_minor_ver = curr_ver_splitted[1] if len(curr_ver_splitted) > 1 else None
+      latest_minor_ver = latest_ver_splitted[1] if len(latest_ver_splitted) > 1 else None
+      if curr_minor_ver is not None and latest_minor_ver is not None:
+        if (not curr_minor_ver.isdigit() or not latest_minor_ver.isdigit()) and curr_minor_ver != latest_minor_ver:
+          return True
+        elif int(curr_minor_ver) + _MAX_MINOR_VERSION_DIFF <= int(latest_minor_ver):
+          return True
+  return False
```
[the remark this answers is not in the document]
Comment (chamikaramj) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Add a TODO to consider patch version later ?

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

**Comment by yifanzou** [comment]
Run Seed Job

**Comment by yifanzou** [comment]
Run Dependency Check

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (yifanzou) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator.py]: Done.

File: .test-infra/jenkins/dependency_check/dependency_check_report_generator_test.py (hunk 12 above)
Comment (yifanzou) [reply .test-infra/jenkins/dependency_check/dependency_check_report_generator_test.py]: Done. Added the unit test into the jenkins job. Fail the job if the unit test did not pass.

**Comment by yifanzou** [comment]
@chamikaramj  PTAL
1. Run the report generator unit tests in Jenkins, and fail the job if test doesn't pass.
2. Squashed commits.

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Pull request merged**